src/Analyzer.Utilities/SyntaxGeneratorExtensions.cs

```diff
@@ -1,66 +1,305 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
-using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Editing;
 
 namespace Analyzer.Utilities
 {
     public static class SyntaxGeneratorExtensions
     {
-        private static readonly OperatorKind[] s_comparisonOperators =
+        /// <summary>
+        /// Creates a declaration for an operator equality overload.
+        /// </summary>
+        /// <param name="generator">
+        /// The <see cref="SyntaxGenerator"/> used to create the declaration.
+        /// </param>
+        /// <param name="containingType">
+        /// A symbol specifying the type of the operands of the comparison operator.
+        /// </param>
+        /// <returns>
+        /// A <see cref="SyntaxNode"/> representing the declaration.
+        /// </returns>
+        public static SyntaxNode OperatorEqualityDeclaration(this SyntaxGenerator generator,
+            INamedTypeSymbol containingType)
         {
-            OperatorKind.Equality,
-            OperatorKind.Inequality,
-            OperatorKind.GreaterThan,
-            OperatorKind.GreaterThanOrEqual,
-            OperatorKind.LessThan,
-            OperatorKind.LessThanOrEqual
-        };
+            var leftArgument = generator.IdentifierName("left");
```

Comment:
Can you please extract all string constants to a fields and share between related methods?